packages/compass-components/src/components/file-input.tsx

```diff
@@ -121,56 +121,59 @@ function FileInput({
   );
 
   return (
-    <div
-      className={cx(
-        { [formItemHorizontalStyles]: variant === Variant.Horizontal },
-        { [formItemVerticalStyles]: variant === Variant.Vertical },
-        { [formItemErrorStyles]: error }
-      )}
-    >
-      <label
-        htmlFor={id}
+    <div className={className}>
+      <div
         className={cx(
-          { [labelHorizontalStyles]: variant === Variant.Horizontal },
-          { [labelErrorStyles]: error }
-        )}
-      >
-        <span>{label}</span>
-        {link && (
-          <a
-            href={link}
-            target="_blank"
-            rel="noreferrer"
-            className={labelIconStyles}
-            data-testid="file-input-link"
-          >
-            
-          </a>
+          { [formItemHorizontalStyles]: variant === Variant.Horizontal },
+          { [formItemVerticalStyles]: variant === Variant.Vertical },
+          { [formItemErrorStyles]: Boolean(error) },
         )}
-      </label>
-      <input
-        ref={inputRef}
-        id={`${id}_file_input`}
-        name={id}
-        type="file"
-        multiple={multi}
-        onChange={onFilesChanged}
-        style={{ display: 'none' }}
-      />
-      <Button
-        id={id}
-        data-testid="file-input-button"
-        className={cx({ [buttonStyles]: true }, { [buttonErrorStyles]: error })}
-        onClick={() => {
-          if (inputRef.current) {
-            inputRef.current.click();
-          }
-        }}
-        title="Select a file"
-        leftGlyph={<Icon glyph="AddFile" title={null} fill="currentColor" />}
       >
-        {buttonText}
-      </Button>
+        <label
+          htmlFor={id}
+          className={cx(
+            { [labelHorizontalStyles]: variant === Variant.Horizontal },
+          )}
+        >
+          <span>{label}</span>
+          {link && !helpText && (
+            <a
+              href={link}
+              target="_blank"
+              rel="noreferrer"
+              className={labelIconStyles}
+              data-testid="file-input-link"
+            >
+              
+            </a>
+          )}
+          {!link && helpText}
```

Comment:
Why having link here disables the helpText? For me it looks like these options are serving a different purpose. Feel like if we want this component to be so generic, this interface might need a bit more refinement

This helpText also seems to play the role of what leafygreen calls description, should we align the prop names?